lib/maplibre-compose-expressions/src/commonMain/kotlin/dev/sargunv/maplibrecompose/expression/dsl/types.kt

```diff
@@ -0,0 +1,372 @@
+    )
+    .asString()
+    .cast()
+}
+
+/**
+ * Asserts that this value is a number.
+ *
+ * In case this expression is not a number, each of the [fallbacks] is evaluated in order until a
+ * number is obtained. If none of the inputs are numbers, the expression is an error.
+ */
+public fun Expression<*>.asNumber(vararg fallbacks: Expression<*>): Expression<FloatValue> =
+  FunctionCall.of("number", this, *fallbacks).cast()
+
+/**
+ * Asserts that this value is a boolean.
+ *
+ * In case this expression is not a boolean, each of the [fallbacks] is evaluated in order until a
+ * boolean is obtained. If none of the inputs are booleans, the expression is an error.
+ */
+public fun Expression<*>.asBoolean(vararg fallbacks: Expression<*>): Expression<BooleanValue> =
+  FunctionCall.of("boolean", this, *fallbacks).cast()
+
+/**
+ * Asserts that this value is a map.
+ *
+ * In case this expression is not a map, each of the [fallbacks] is evaluated in order until a map
+ * is obtained. If none of the inputs are maps, the expression is an error.
+ */
+public fun Expression<*>.asMap(vararg fallbacks: Expression<*>): Expression<MapValue<*>> =
+  FunctionCall.of("object", this, *fallbacks).cast()
+
+/**
+ * Returns a collator for use in locale-dependent comparison operations. The [caseSensitive] and
+ * [diacriticSensitive] options default to `false`. The [locale] argument specifies the IETF
+ * language tag of the locale to use. If none is provided, the default locale is used. If the
+ * requested locale is not available, the collator will use a system-defined fallback locale. Use
+ * [resolvedLocale] to test the results of locale fallback behavior.
+ */
+public fun collator(
+  caseSensitive: Expression<BooleanValue>? = null,
+  diacriticSensitive: Expression<BooleanValue>? = null,
+  locale: Expression<StringValue>? = null,
+): Expression<CollatorValue> =
+  FunctionCall.of(
+      "collator",
+      Options.build(
+        fun MutableMap<String, Expression<*>>.() {
+          caseSensitive?.let { put("case-sensitive", it) }
+          diacriticSensitive?.let { put("diacritic-sensitive", it) }
+          locale?.let { put("locale", it) }
+        }
+      ),
+    )
+    .cast()
+
+/**
+ * Returns a formatted string for displaying mixed-format text in the `textField` property (see
+ * [SymbolLayer][dev.sargunv.maplibrecompose.compose.layer.SymbolLayer]). The input may contain a
+ * string literal or expression, including an [image] expression.
+ *
+ * Example:
+ * ```
+ * format(
+ *   span(
+ *     feature.get("name").asString().substring(const(0), const(1)).uppercase(),
+ *     textScale = const(1.5f),
+ *   ),
+ *   span(feature.get("name").asString().substring(const(1)))
+ * )
+ * ```
+ *
+ * Capitalizes the first letter of the features' property "name" and formats it to be extra-large,
+ * the rest of the name is written normally.
+ */
+public fun format(vararg spans: FormatSpan): Expression<FormattedValue> =
```

Comment:
I'd expect the `format` stuff to be either in its own file or in the file with the other string-related functions rather than in the file for "casting".

Same with `image`.

I mean, they are not really related, right?

Reply:
makes sense. I mostly just used the existing `// region` declarations to split these files